Update ANGLE to 2024-03-25 (3ed8c5f4314708d3078950620b135722eada60bf)

kkinnunen-apple opened the pull request:
#### 60f80224d2c98fea3d208c385f050a662f34ae32
<pre>
Update ANGLE to 2024-03-25 (3ed8c5f4314708d3078950620b135722eada60bf)
<a href="https://bugs.webkit.org/show_bug.cgi?id=272023">https://bugs.webkit.org/show_bug.cgi?id=272023</a>
<a href="https://rdar.apple.com/problem/125776346">rdar://problem/125776346</a>

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline f431641a948660f5e1709aa7cd89b16ccf93f1de..3ed8c5f4314708d3078950620b135722eada60bf --pretty=%h %s
3ed8c5f431 Android: workaround broken run-as due to /data permissions
96f443295f Revert &quot;Remove few redundant ImmutableString to std::string conversions&quot;
ee02014d87 Selectively wait for LinkSubTasks
16fef70f84 Vulkan: Refactor imageless framebuffer creation
1bd82319c7 Add RenderTargetWgpu
a53bd62a90 Disable MSRTSS on QCOM
33a09305f1 Metal: Remove work texture and work buffer from ContextMtl
0197826bba Revert &quot;Add conversion operator from ImmutableString to std::string&quot;
9ae4a2f4b7 Roll Chromium from babb076716bc to 532d52d6ecb3 (627 revisions)
ad013650bb Revert &quot;Rename LinkSubTask -&gt; PostLinkTask&quot;
8899e18dc6 Add skip conditions for a few tests
f3a819f1bc Context: Limit max texture size for ANGLE captures
6c37973311 Add more helper methods for textures.
a69c56fa88 Pin build until Skia is able to build with ToT abseil.
da8fa1b615 Roll Chromium from 42826620a07f to babb076716bc (4096 revisions)
07137f571a Add ImageHelper to TextureWgpu
8c0dae388b Add conversion operator from ImmutableString to std::string
ed97adba3d Add more RGB-to-RGBA byte loading functions
00eb6edba0 Rename LinkSubTask -&gt; PostLinkTask
042b430c63 Support the wgpu backend on Windows
96f66089c2 Tests: Update color values for Unsized/Unsigned in texture tests
f8b185771b Set the Dawn proc addresses at in DisplayWgpu initialization
7047d8205c win-trace (capture_replay_tests): cleanup after replay
21b5c32113 Pin abseil-cpp until Skia is able to build with ToT abseil.
a3dff4ed6e third_party/clspv: Disable the warnings on build
efd41bd207 Vulkan: Rename ResourceVk.* to vk_resource.*
e3aac00be1 Vulkan: only request OPAQUE compositeAlpha on Android if no alpha
914fe61bc6 Vulkan: Rename RendererVk.* to vk_renderer.*
1fc548c115 Vulkan: enable recordable bit for RGB8
ebb94b807f Remove few redundant ImmutableString to std::string conversions
7cb518bc22 Reset shader images modified during trace
7506a0cdc0 Tests: update QDC expectations after crrev.com/c/5372826
2f934a47e9 libstdc++: replace std::powf with std:pow
6e6d6b2983 Roll vulkan-deps from fa1e68dabb91 to ba66ec69216d (8 revisions)
d2cef82a8f Vulkan: Use fragment shading rate access flags
6d5690036a Vulkan: Deduplicate merge to Renderer&apos;s pipeline cache
490ff869e9 Always redeclare clip/cull distance built-ins
f4ecbe6f13 Use Dawn&apos;s abseil build files when wgpu is enabled.
afbfebcaaf Reland &quot;Manual roll VK-GL-CTS from 87353392d2d2 to ec9827528085 (9 revisions)&quot;
92b8fc6889 Revert &quot;Manual roll VK-GL-CTS from 87353392d2d2 to ec9827528085 (9 revisions)&quot;
f8021cfa91 Manual roll vulkan-deps from 91aab7b7e25f to fa1e68dabb91 (27 revisions)
8080a736f1 Manual roll VK-GL-CTS from 87353392d2d2 to ec9827528085 (9 revisions)
21d124c4bf Vulkan: Remove support for pipeline cache control
553e3c8038 Vulkan: Async compile pipelines with different surface rotations
58065d0766 Vulkan: Add test that destroys view after the image
c5f7bbeb11 Make ImmutableString::beginsWith constexpr
21ef298e33 Consider textures without an attached Buffer as incomplete
60aaf4a096 Vulkan: Move renderer to namespace vk
b3ab67d32b tests: Remove unnecessary .get() from RAII objects
d4d3478145 Multisampling support check: sampleCounts &gt; 1 and createFlags
2dc45de80d Trace tests: check and log zlib crc32 on decompress failure
7220307bb2 Conditionally support EGL_OPENGL_API
b5d92511b4 Trace tests: log z_stream on decompress failure
c3fdc81189 Raise the WebGL texture size limit on Android &gt;= 14
e5cb7f1f5c Vulkan: Fix access to inactive attributes
533fa5cedc Metal: Update few asserts when resolving FS output
9bae585946 Vulkan: Add blend factors to allow dithering to work
142c05b543 Build OpenCL CTS on ANGLE Linux and Android standalone bots
afed1224de CL: Add CTS build to ANGLE
cd90294be8 OpenCL/Vulkan: Add initial CL Event creation
66bc9cfa00 OpenCL: Fix mCallback in linkProgram ctor
17facd2b06 Fix CreateDirectory name clash with Windows headers.
18fa02bebf Rewrite exprs using separated decl variables
cd220fa9ec Roll vulkan-deps from 6e41ca22e87b to 91aab7b7e25f (5 revisions)
5673346918 Roll Chromium from 63f2e96c7e8a to 42826620a07f (605 revisions)
f7de39b6ae Make IntermRebuild available for all backends
065b333cad OpenCL/Vulkan: Add initial CL Buffer routines
aea2abc1ad Vulkan: Input attachment requires both texturable and renderable
da3baf096b Manual roll VK-GL-CTS from 1918ab4d4806 to 87353392d2d2 (15 revisions)
b744ee7f16 Trace tests: extract debug files to (CAS) output
529a2fe904 Roll vulkan-deps from 3b14ca63bd7b to 6e41ca22e87b (21 revisions)
18e00a2989 Roll Chromium from d4320bd12d3a to 63f2e96c7e8a (593 revisions)
a339585e1a Vulkan: Add AtomicShared type
a55f91b566 Switch Pixel 6 experimental bot to Android U
bc633ad708 OpenCL/Vulkan: Implement compile and link routines
99c2157b1d Metal: Add Quyen as OWNER
d964482310 Make ANGLE by default built to system partition on Android
b9c6cf8ecb Metal: Stop blit encoder after render encoder
c9c7c4d74a Trace tests: save debug files when gz decompression fails
9fee915631 OpenCL/Vulkan: Add initial CL Kernel routines
7f3bb2d80f Roll Chromium from 93f3c55ed974 to d4320bd12d3a (756 revisions)
f16eea308a Vulkan: Enable QCOM foveated rendering extensions
e904e37ba3 Vulkan: Enable imageless framebuffer on Samsung drivers
11a2d27f32 Check array index against unsigned array size
26da3174dd Make 2024-03-05 changes compile with clang 15 pt.2
90ae6cbe39 Avoid assert at main prototype when monomorphizing
0f110098cc Avoid assert with multiple memory qualifiers
c55c8ad21c extension XML cleanup
6ba49977d8 CL: Update OpenCL Headers
1452c19542 Android: Add Qualcomm Mobile Reference Device support
a40eeaa9e0 Roll Chromium from 15a5ccdeffb7 to 93f3c55ed974 (614 revisions)
ecaefce00c Vulkan: Disable optimizeWithLoadOp if there is unresolve
4667201495 Vulkan: Add test for midRenderPass clear for MSRTT
aba3705ba7 Vulkan: Completely remove egl::Display from RendererVk
7e065b6f4d Fix SRV and RTV confliction
74af31adca GL: Add ClearsWithGapsNeedFlush workaround
4a5b9307be android_helper: support angle_deqp_egl_tests
49abf72f61 Roll vulkan-deps from 9cd617cb0454 to 3b14ca63bd7b (6 revisions)
dcc79a2764 Roll Chromium from 632158ced47e to 15a5ccdeffb7 (1166 revisions)
91ddf851c4 Vulkan: support QCOM foveated rendering extensions
39f29f65c4 Ensure unary math op parse to an node on error
c71de8688c Add workaround for ext dynamic state on Win/Intel
4e9fbb36f2 Metal: Remove AccessField(.., ImmutableString)
e38cf95a58 Metal: Release prov. vertex buffers on event set
51702d791d Make 2024-03-05 changes compile with clang 15
d76505b8c3 Manual roll vulkan-deps from 12f9cddb3ff7 to 9cd617cb0454 (42 revisions)
b2773c110f Vulkan: Bug fix in immutable sampler pipeline layout recreation
3c08d69612 CL: Add DEVICE_NOT_FOUND case for context creation
f044aaf821 Vulkan: Create instance/device without access to Display
47cd0529f1 Fix assert invoking #line during macro invocation
27423bffff Metal: Generate names for rewritten inputs
2ad7b23b13 Add a missing #include.
545e3f6e11 Vulkan: Decouple RendererVk from egl::BlobCache
95294b2468 Android: Add Galaxy S22 support (Xclipse)
5678ad09aa Roll Chromium from 43d81add625d to 632158ced47e (570 revisions)
0ad73958dc Deduplicate and fix ConstStrLen implementations
258b751f57 OpenCL/Vulkan: Fix processedOptions whitespace
b978974d98 Update frontend support for QCOM foveated extensions
3fa8d578ad Make appendDecimal use the last char of the buffer
39040b0b89 Vulkan: Decouple RendererVk from EGL attributes
4e6fe5e0db Vulkan: Cache ImageLoadContext in context
871a309c72 Fix layout(index=) parse assert on es 100 shaders
ec6d628863 egl: Add logic to select preferred display
799997d427 Roll Chromium from 40412b90c691 to 43d81add625d (324 revisions)
fc440afa62 Vulkan: Move DS builder class to Vk utils
f85b6970a9 OpenCL/Vulkan: Implement program get[Build]Info
0ed0de4f0b OpenCL/Vulkan: Add initial program build support
1c2d2417c9 Bugfix in CreateWithEGLConfig1010102Support test
f26c8d0874 Roll VK-GL-CTS from d023c17ac299 to 1918ab4d4806 (13 revisions)
21381f5e1c Roll Chromium from 6b34297e693d to 40412b90c691 (533 revisions)
2ee295b475 Vulkan: Add per-level image update tracker
1ceddbf697 OpenCL/Vulkan: Add createProgram routines
f7cd1c5606 Tests: Add Toca Life World trace
56a291e819 Rework external image capture
8142dde7f4 Tests: Add Pokemon Masters Ex trace
b45b350ade Add skip for Pokemon Masters Ex validation warning
69f5e9ca60 Roll vulkan-deps from f43c5512f6d7 to 12f9cddb3ff7 (6 revisions)
19e725e49c Roll Chromium from 579e74402476 to 6b34297e693d (578 revisions)
4d36224267 Vulkan: Remove call to angle::GetSystemInfo()
cdf6220c28 Reland &quot;Vulkan: Feature addition for QCOM foveated rendering extensions&quot;
a971e5b42e Account for zero vector axes in Mat4::Rotate(...)
434a5b0170 Fix #2 upload_results_to_perf_dashboard usage
057db6ef57 Add ANGLE experimental S22 build and test
f8dac42e95 Fix upload_results_to_perf_dashboard usage
dbbcf33eeb Roll vulkan-deps from 28960bf4a098 to f43c5512f6d7 (13 revisions)
d334a6f265 Roll Chromium from cc3c5664ec19 to 579e74402476 (619 revisions)
a627dd8976 Revert &quot;Vulkan: Feature addition for QCOM foveated rendering extensions&quot;
6eaaad7c60 Create ImageHelper.
75c8ef1c63 Update cached component type masks on attachment redefinitions
6f2daf0588 Context: Limit max vtx uniform vectors to 256 during capture
2fb425d284 Roll vulkan-deps from dd6c2371c85d to 28960bf4a098 (10 revisions)
b0215166ed Roll SwiftShader from 0f69b790c7a4 to bbe6452b420c (1 revision)
9100f2ec79 Roll Chromium from 16b5225bad88 to cc3c5664ec19 (580 revisions)
f0af4730d9 Vulkan: Catch misuse of AddToPNextChain
72cf9915f5 Vulkan: Feature addition for QCOM foveated rendering extensions
0afcac60ed Handle count = 0 in DrawElementsIndirect
3c517e457a Vulkan: Process ClearEmulatedChannels update first
38cc4cf099 Vulkan: Update flushStagedUpdate to use switchcase
58c20052bb Fix build error when git history not fully available
d354c4dca1 Roll VK-GL-CTS from d15e5faec700 to d023c17ac299 (1 revision)
425be99db6 Roll vulkan-deps from 602ab4120d74 to dd6c2371c85d (8 revisions)
1fe63fecab Roll SwiftShader from eb75201a4e03 to 0f69b790c7a4 (1 revision)
834ca37fa6 Roll Chromium from b54ff9b1d5ed to 16b5225bad88 (644 revisions)
acba61cb3e Fix Vulkan driver version for Win/Intel
c758dc03c4 GL: Adjust disableRenderSnorm condition
6a88437dc1 Roll VK-GL-CTS from cf5313984f57 to d15e5faec700 (1 revision)
db88630858 Roll VK-GL-CTS from c402aa4fc1f1 to cf5313984f57 (7 revisions)
bd1b918a5a Roll vulkan-deps from 004d9803b30c to 602ab4120d74 (6 revisions)
33a3395599 Roll Chromium from cc824ffe820c to b54ff9b1d5ed (616 revisions)
5a4bfd61fd Metal: Separate struct definition from function return
7f29c70360 Roll vulkan-deps from b040470c0fde to 004d9803b30c (5 revisions)
b06dbffd1c Roll Chromium from 5f0b8ba66cd4 to cc824ffe820c (635 revisions)
cd63c5d477 Fix build failures targetting iOS 17.4
8c503c1b05 Add skip for 07753 validation error in trace
bcf814fda5 Vulkan: Constrain the dependency on ContextVk in BufferHelper
f43b9f87cf Roll vulkan-deps from 5fa0abb9413b to b040470c0fde (5 revisions)
0f6386a82e Roll Chromium from 98827507560a to 5f0b8ba66cd4 (624 revisions)
f546983cc8 Add test and skip for 07753 validation error
8346addbd0 Contain X11 includes and free usage of common terms
1ee04579b4 Metal: Re-enable asm inejction into loops on MacOS 12+
21c0d31cd6 OpenCL: Only build clspv for Vk backend
b7bacdb746 GL: Generate mipmaps through draw calls on Pixel7/Pixel8.
2b1ef00ad5 Metal: Fix validation for anonymous struct arrays
19e21b1e0b OpenCL/Vulkan: Add initial support for cmdQueue
f4d5644c9f Instantiate dawn backend in angle_end2end_tests
b7d0a18bb1 Roll vulkan-deps from 13783d616289 to 5fa0abb9413b (10 revisions)
9f16b585cb Roll Chromium from 2905059a5737 to 98827507560a (299 revisions)
e17dd5a408 Roll vulkan-deps from 063ea20a64fc to 13783d616289 (25 revisions)
e3badd04cc Roll Chromium from 9d4a35b46e1e to 2905059a5737 (725 revisions)
e04b7c7392 Vulkan: Expand feature to enable sample usage for all AHBs
e08e82b6de CL: On kernel arg validation use right sizes
9d453e579e Fix ASSERT in non-global precise var decls
bde26cc4de Roll VK-GL-CTS from b9ec0d4bdf99 to c402aa4fc1f1 (13 revisions)
372876879b Android: support running angle_unittests via android_helper
a6616081f2 Add missing include
af56ca61bb OpenCL/Vulkan: Initial support for context
298abbc156 Roll Chromium from 29bec8631d2f to 9d4a35b46e1e (1220 revisions)
3ca8befb24 Vulkan: Handle multi-context apps in pipeline cache graphs
6607a2b98d Vulkan: Add support for VK_EXT_vertex_input_dynamic_state
bff0b1e43d Change enum value for webgpu to unused value.
6d4706bfb9 WGPU: Add a angle_dawn_dir build override.
aa244358af Reland &quot;Vulkan: Get rid of X11 include in DisplayVkXcb.cpp&quot;
6d589ff6a3 Trace perf: support custom thermal throttling for tests
ec2603d69f Fix build in absence of SSE support
b380ed1f98 Vulkan: Add EGL_ANGLE_global_fence_sync
40dfb3a8bd Fix length() translation for clip/cull distance arrays
d6ceac9159 Metal: Add support for binding slices to images
195c142d7e GLSL test for side effects in prune-able loop
e8a3493f80 Initialize DisplayWgpu
2cae27c296 Vulkan: Enable the doubleDepthBiasConstantFactor feature on NV
197beb4de8 Metal: Crash if for loop body is optimized away
dbc6bd9d4e Reland &quot;Vulkan: Fix alignment issues with SecondaryCommandBuffer&quot;
c673c83758 OpenCL/Vulkan: Initial support for platform/device
b8e56d5d6d Fix an assert when overwriting TexImage binding
243f8ad99f Revert &quot;Vulkan: Fix alignment issues with SecondaryCommandBuffer&quot;
7490ad4d79 Roll vulkan-deps from 2cedf06e4cdf to 063ea20a64fc (7 revisions)
d8340c15c7 Roll Chromium from b650d7fcd665 to 29bec8631d2f (631 revisions)
e53270c9ca Vulkan: Fix alignment issues with SecondaryCommandBuffer
e45b2fd89d Vulkan: Implement ANGLE_translated_shader_source
d9665098a3 Do not use hardcoded &quot;.cr.so&quot; extension for android component builds
9d344b5c82 Uniform block reference in constuctors crash
ebc151d514 Roll vulkan-deps from 4985acbd814d to 2cedf06e4cdf (12 revisions)
2cbf6613c1 Roll Chromium from 3009d13b1e1a to b650d7fcd665 (3044 revisions)
cb7d3cc206 Treat clip/cull distance built-ins as having side effects
e784b1ec82 Manual roll dawn
239e8caa44 Capture/Replay: Disallow concurrent ninja processes.
9fd5167e1f Roll vulkan-deps from 3834da2004ec to 4985acbd814d (77 revisions)
eaddd3baa5 Vulkan: use linear chroma filter for ycbcr by default
1d752a10a7 Roll Chromium from c1ca24b91ed5 to 3009d13b1e1a (567 revisions)
275e6f4fc5 D3D: Add multiplanar support to d3d11 glTexSubImage2D
e489dac03a Allow BGRA -&gt; RGBA for glCopyTex[Sub]Image
5d9abeca4a Revert &quot;Suppress VUID-VkGraphicsPipelineCreateInfo-dynamicRendering-06576&quot;
56e5fa804b Mark bison deps as not shipped
58ccdab977 Roll VK-GL-CTS from b3344240e7fc to b9ec0d4bdf99 (8 revisions)
c3d06480e5 Add dawn to IGNORED_DIRECTORIES.
73fa1b08d0 Use -fno-define-target-os-macros for libpng
3ad163d091 Vulkan: Don&apos;t attach format features 2 version of AHB structure
475784f5a6 suppress VUID-VkPipelineVertexInputStateCreateInfo-pNext-pNext
c603a4f199 Don&apos;t perf warn about ETC1-&gt;ETC2 emulation as it is efficient

ef78e57015 Revert &quot;Vulkan: disable warmUpPipelineCacheAtLink for Venus&quot;
Canonical link: <a href="https://commits.webkit.org/276989@main">https://commits.webkit.org/276989@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0cfa05aca038f8ef0893284262a61af09af8cbaa

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46266 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25400 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48854 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48939 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42307 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29756 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22858 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37795 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46844 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22487 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39903 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19009 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19866 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41001 "Passed tests") | [❌ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4310 "Hash 0cfa05ac for PR 26717 does not build (failure)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42581 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41356 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50751 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/21267 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17763 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44998 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22562 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43913 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10258 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/22929 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22258 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
```diff
@@ -328,8 +328,6 @@
 		31CDFFFD249184B000486F27 /* imageformats.h in Headers */ = {isa = PBXBuildFile; fileRef = 5CCD598B2284FC400018F2D8 /* imageformats.h */; };
 		31CDFFFE249184B000486F27 /* copyimage.h in Headers */ = {isa = PBXBuildFile; fileRef = 5CCD59862284FC400018F2D8 /* copyimage.h */; };
 		31CDFFFF249184B000486F27 /* generatemip.h in Headers */ = {isa = PBXBuildFile; fileRef = 5CCD598C2284FC400018F2D8 /* generatemip.h */; };
-		3A084D472B8C317B00374F23 /* SeparateStructFromFunctionDeclarations.h in Headers */ = {isa = PBXBuildFile; fileRef = 3A084D452B8C317B00374F23 /* SeparateStructFromFunctionDeclarations.h */; };
```
These files support `5a4bfd61fd Metal: Separate struct definition from function return`, so It's curious that we're reimporting that change.

djg reviewed (approved):
:shipit:

kkinnunen-apple committed:
Update ANGLE to 2024-03-25 (3ed8c5f4314708d3078950620b135722eada60bf)
https://bugs.webkit.org/show_bug.cgi?id=272023
rdar://problem/125776346

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline f431641a948660f5e1709aa7cd89b16ccf93f1de..3ed8c5f4314708d3078950620b135722eada60bf --pretty=%h %s
3ed8c5f431 Android: workaround broken run-as due to /data permissions
96f443295f Revert "Remove few redundant ImmutableString to std::string conversions"
ee02014d87 Selectively wait for LinkSubTasks
16fef70f84 Vulkan: Refactor imageless framebuffer creation
1bd82319c7 Add RenderTargetWgpu
a53bd62a90 Disable MSRTSS on QCOM
33a09305f1 Metal: Remove work texture and work buffer from ContextMtl
0197826bba Revert "Add conversion operator from ImmutableString to std::string"
9ae4a2f4b7 Roll Chromium from babb076716bc to 532d52d6ecb3 (627 revisions)
ad013650bb Revert "Rename LinkSubTask -> PostLinkTask"
8899e18dc6 Add skip conditions for a few tests
f3a819f1bc Context: Limit max texture size for ANGLE captures
6c37973311 Add more helper methods for textures.
a69c56fa88 Pin build until Skia is able to build with ToT abseil.
da8fa1b615 Roll Chromium from 42826620a07f to babb076716bc (4096 revisions)
07137f571a Add ImageHelper to TextureWgpu
8c0dae388b Add conversion operator from ImmutableString to std::string
ed97adba3d Add more RGB-to-RGBA byte loading functions
00eb6edba0 Rename LinkSubTask -> PostLinkTask
042b430c63 Support the wgpu backend on Windows
96f66089c2 Tests: Update color values for Unsized/Unsigned in texture tests
f8b185771b Set the Dawn proc addresses at in DisplayWgpu initialization
7047d8205c win-trace (capture_replay_tests): cleanup after replay
21b5c32113 Pin abseil-cpp until Skia is able to build with ToT abseil.
a3dff4ed6e third_party/clspv: Disable the warnings on build
efd41bd207 Vulkan: Rename ResourceVk.* to vk_resource.*
e3aac00be1 Vulkan: only request OPAQUE compositeAlpha on Android if no alpha
914fe61bc6 Vulkan: Rename RendererVk.* to vk_renderer.*
1fc548c115 Vulkan: enable recordable bit for RGB8
ebb94b807f Remove few redundant ImmutableString to std::string conversions
7cb518bc22 Reset shader images modified during trace
7506a0cdc0 Tests: update QDC expectations after crrev.com/c/5372826
2f934a47e9 libstdc++: replace std::powf with std:pow
6e6d6b2983 Roll vulkan-deps from fa1e68dabb91 to ba66ec69216d (8 revisions)
d2cef82a8f Vulkan: Use fragment shading rate access flags
6d5690036a Vulkan: Deduplicate merge to Renderer's pipeline cache
490ff869e9 Always redeclare clip/cull distance built-ins
f4ecbe6f13 Use Dawn's abseil build files when wgpu is enabled.
afbfebcaaf Reland "Manual roll VK-GL-CTS from 87353392d2d2 to ec9827528085 (9 revisions)"
92b8fc6889 Revert "Manual roll VK-GL-CTS from 87353392d2d2 to ec9827528085 (9 revisions)"
f8021cfa91 Manual roll vulkan-deps from 91aab7b7e25f to fa1e68dabb91 (27 revisions)
8080a736f1 Manual roll VK-GL-CTS from 87353392d2d2 to ec9827528085 (9 revisions)
21d124c4bf Vulkan: Remove support for pipeline cache control
553e3c8038 Vulkan: Async compile pipelines with different surface rotations
58065d0766 Vulkan: Add test that destroys view after the image
c5f7bbeb11 Make ImmutableString::beginsWith constexpr
21ef298e33 Consider textures without an attached Buffer as incomplete
60aaf4a096 Vulkan: Move renderer to namespace vk
b3ab67d32b tests: Remove unnecessary .get() from RAII objects
d4d3478145 Multisampling support check: sampleCounts > 1 and createFlags
2dc45de80d Trace tests: check and log zlib crc32 on decompress failure
7220307bb2 Conditionally support EGL_OPENGL_API
b5d92511b4 Trace tests: log z_stream on decompress failure
c3fdc81189 Raise the WebGL texture size limit on Android >= 14
e5cb7f1f5c Vulkan: Fix access to inactive attributes
533fa5cedc Metal: Update few asserts when resolving FS output
9bae585946 Vulkan: Add blend factors to allow dithering to work
142c05b543 Build OpenCL CTS on ANGLE Linux and Android standalone bots
afed1224de CL: Add CTS build to ANGLE
cd90294be8 OpenCL/Vulkan: Add initial CL Event creation
66bc9cfa00 OpenCL: Fix mCallback in linkProgram ctor
17facd2b06 Fix CreateDirectory name clash with Windows headers.
18fa02bebf Rewrite exprs using separated decl variables
cd220fa9ec Roll vulkan-deps from 6e41ca22e87b to 91aab7b7e25f (5 revisions)
5673346918 Roll Chromium from 63f2e96c7e8a to 42826620a07f (605 revisions)
f7de39b6ae Make IntermRebuild available for all backends
065b333cad OpenCL/Vulkan: Add initial CL Buffer routines
aea2abc1ad Vulkan: Input attachment requires both texturable and renderable
da3baf096b Manual roll VK-GL-CTS from 1918ab4d4806 to 87353392d2d2 (15 revisions)
b744ee7f16 Trace tests: extract debug files to (CAS) output
529a2fe904 Roll vulkan-deps from 3b14ca63bd7b to 6e41ca22e87b (21 revisions)
18e00a2989 Roll Chromium from d4320bd12d3a to 63f2e96c7e8a (593 revisions)
a339585e1a Vulkan: Add AtomicShared type
a55f91b566 Switch Pixel 6 experimental bot to Android U
bc633ad708 OpenCL/Vulkan: Implement compile and link routines
99c2157b1d Metal: Add Quyen as OWNER
d964482310 Make ANGLE by default built to system partition on Android
b9c6cf8ecb Metal: Stop blit encoder after render encoder
c9c7c4d74a Trace tests: save debug files when gz decompression fails
9fee915631 OpenCL/Vulkan: Add initial CL Kernel routines
7f3bb2d80f Roll Chromium from 93f3c55ed974 to d4320bd12d3a (756 revisions)
f16eea308a Vulkan: Enable QCOM foveated rendering extensions
e904e37ba3 Vulkan: Enable imageless framebuffer on Samsung drivers
11a2d27f32 Check array index against unsigned array size
26da3174dd Make 2024-03-05 changes compile with clang 15 pt.2
90ae6cbe39 Avoid assert at main prototype when monomorphizing
0f110098cc Avoid assert with multiple memory qualifiers
c55c8ad21c extension XML cleanup
6ba49977d8 CL: Update OpenCL Headers
1452c19542 Android: Add Qualcomm Mobile Reference Device support
a40eeaa9e0 Roll Chromium from 15a5ccdeffb7 to 93f3c55ed974 (614 revisions)
ecaefce00c Vulkan: Disable optimizeWithLoadOp if there is unresolve
4667201495 Vulkan: Add test for midRenderPass clear for MSRTT
aba3705ba7 Vulkan: Completely remove egl::Display from RendererVk
7e065b6f4d Fix SRV and RTV confliction
74af31adca GL: Add ClearsWithGapsNeedFlush workaround
4a5b9307be android_helper: support angle_deqp_egl_tests
49abf72f61 Roll vulkan-deps from 9cd617cb0454 to 3b14ca63bd7b (6 revisions)
dcc79a2764 Roll Chromium from 632158ced47e to 15a5ccdeffb7 (1166 revisions)
91ddf851c4 Vulkan: support QCOM foveated rendering extensions
39f29f65c4 Ensure unary math op parse to an node on error
c71de8688c Add workaround for ext dynamic state on Win/Intel
4e9fbb36f2 Metal: Remove AccessField(.., ImmutableString)
e38cf95a58 Metal: Release prov. vertex buffers on event set
51702d791d Make 2024-03-05 changes compile with clang 15
d76505b8c3 Manual roll vulkan-deps from 12f9cddb3ff7 to 9cd617cb0454 (42 revisions)
b2773c110f Vulkan: Bug fix in immutable sampler pipeline layout recreation
3c08d69612 CL: Add DEVICE_NOT_FOUND case for context creation
f044aaf821 Vulkan: Create instance/device without access to Display
47cd0529f1 Fix assert invoking #line during macro invocation
27423bffff Metal: Generate names for rewritten inputs
2ad7b23b13 Add a missing #include.
545e3f6e11 Vulkan: Decouple RendererVk from egl::BlobCache
95294b2468 Android: Add Galaxy S22 support (Xclipse)
5678ad09aa Roll Chromium from 43d81add625d to 632158ced47e (570 revisions)
0ad73958dc Deduplicate and fix ConstStrLen implementations
258b751f57 OpenCL/Vulkan: Fix processedOptions whitespace
b978974d98 Update frontend support for QCOM foveated extensions
3fa8d578ad Make appendDecimal use the last char of the buffer
39040b0b89 Vulkan: Decouple RendererVk from EGL attributes
4e6fe5e0db Vulkan: Cache ImageLoadContext in context
871a309c72 Fix layout(index=) parse assert on es 100 shaders
ec6d628863 egl: Add logic to select preferred display
799997d427 Roll Chromium from 40412b90c691 to 43d81add625d (324 revisions)
fc440afa62 Vulkan: Move DS builder class to Vk utils
f85b6970a9 OpenCL/Vulkan: Implement program get[Build]Info
0ed0de4f0b OpenCL/Vulkan: Add initial program build support
1c2d2417c9 Bugfix in CreateWithEGLConfig1010102Support test
f26c8d0874 Roll VK-GL-CTS from d023c17ac299 to 1918ab4d4806 (13 revisions)
21381f5e1c Roll Chromium from 6b34297e693d to 40412b90c691 (533 revisions)
2ee295b475 Vulkan: Add per-level image update tracker
1ceddbf697 OpenCL/Vulkan: Add createProgram routines
f7cd1c5606 Tests: Add Toca Life World trace
56a291e819 Rework external image capture
8142dde7f4 Tests: Add Pokemon Masters Ex trace
b45b350ade Add skip for Pokemon Masters Ex validation warning
69f5e9ca60 Roll vulkan-deps from f43c5512f6d7 to 12f9cddb3ff7 (6 revisions)
19e725e49c Roll Chromium from 579e74402476 to 6b34297e693d (578 revisions)
4d36224267 Vulkan: Remove call to angle::GetSystemInfo()
cdf6220c28 Reland "Vulkan: Feature addition for QCOM foveated rendering extensions"
a971e5b42e Account for zero vector axes in Mat4::Rotate(...)
434a5b0170 Fix #2 upload_results_to_perf_dashboard usage
057db6ef57 Add ANGLE experimental S22 build and test
f8dac42e95 Fix upload_results_to_perf_dashboard usage
dbbcf33eeb Roll vulkan-deps from 28960bf4a098 to f43c5512f6d7 (13 revisions)
d334a6f265 Roll Chromium from cc3c5664ec19 to 579e74402476 (619 revisions)
a627dd8976 Revert "Vulkan: Feature addition for QCOM foveated rendering extensions"
6eaaad7c60 Create ImageHelper.
75c8ef1c63 Update cached component type masks on attachment redefinitions
6f2daf0588 Context: Limit max vtx uniform vectors to 256 during capture
2fb425d284 Roll vulkan-deps from dd6c2371c85d to 28960bf4a098 (10 revisions)
b0215166ed Roll SwiftShader from 0f69b790c7a4 to bbe6452b420c (1 revision)
9100f2ec79 Roll Chromium from 16b5225bad88 to cc3c5664ec19 (580 revisions)
f0af4730d9 Vulkan: Catch misuse of AddToPNextChain
72cf9915f5 Vulkan: Feature addition for QCOM foveated rendering extensions
0afcac60ed Handle count = 0 in DrawElementsIndirect
3c517e457a Vulkan: Process ClearEmulatedChannels update first
38cc4cf099 Vulkan: Update flushStagedUpdate to use switchcase
58c20052bb Fix build error when git history not fully available
d354c4dca1 Roll VK-GL-CTS from d15e5faec700 to d023c17ac299 (1 revision)
425be99db6 Roll vulkan-deps from 602ab4120d74 to dd6c2371c85d (8 revisions)
1fe63fecab Roll SwiftShader from eb75201a4e03 to 0f69b790c7a4 (1 revision)
834ca37fa6 Roll Chromium from b54ff9b1d5ed to 16b5225bad88 (644 revisions)
acba61cb3e Fix Vulkan driver version for Win/Intel
c758dc03c4 GL: Adjust disableRenderSnorm condition
6a88437dc1 Roll VK-GL-CTS from cf5313984f57 to d15e5faec700 (1 revision)
db88630858 Roll VK-GL-CTS from c402aa4fc1f1 to cf5313984f57 (7 revisions)
bd1b918a5a Roll vulkan-deps from 004d9803b30c to 602ab4120d74 (6 revisions)
33a3395599 Roll Chromium from cc824ffe820c to b54ff9b1d5ed (616 revisions)
5a4bfd61fd Metal: Separate struct definition from function return
7f29c70360 Roll vulkan-deps from b040470c0fde to 004d9803b30c (5 revisions)
b06dbffd1c Roll Chromium from 5f0b8ba66cd4 to cc824ffe820c (635 revisions)
cd63c5d477 Fix build failures targetting iOS 17.4
8c503c1b05 Add skip for 07753 validation error in trace
bcf814fda5 Vulkan: Constrain the dependency on ContextVk in BufferHelper
f43b9f87cf Roll vulkan-deps from 5fa0abb9413b to b040470c0fde (5 revisions)
0f6386a82e Roll Chromium from 98827507560a to 5f0b8ba66cd4 (624 revisions)
f546983cc8 Add test and skip for 07753 validation error
8346addbd0 Contain X11 includes and free usage of common terms
1ee04579b4 Metal: Re-enable asm inejction into loops on MacOS 12+
21c0d31cd6 OpenCL: Only build clspv for Vk backend
b7bacdb746 GL: Generate mipmaps through draw calls on Pixel7/Pixel8.
2b1ef00ad5 Metal: Fix validation for anonymous struct arrays
19e21b1e0b OpenCL/Vulkan: Add initial support for cmdQueue
f4d5644c9f Instantiate dawn backend in angle_end2end_tests
b7d0a18bb1 Roll vulkan-deps from 13783d616289 to 5fa0abb9413b (10 revisions)
9f16b585cb Roll Chromium from 2905059a5737 to 98827507560a (299 revisions)
e17dd5a408 Roll vulkan-deps from 063ea20a64fc to 13783d616289 (25 revisions)
e3badd04cc Roll Chromium from 9d4a35b46e1e to 2905059a5737 (725 revisions)
e04b7c7392 Vulkan: Expand feature to enable sample usage for all AHBs
e08e82b6de CL: On kernel arg validation use right sizes
9d453e579e Fix ASSERT in non-global precise var decls
bde26cc4de Roll VK-GL-CTS from b9ec0d4bdf99 to c402aa4fc1f1 (13 revisions)
372876879b Android: support running angle_unittests via android_helper
a6616081f2 Add missing include
af56ca61bb OpenCL/Vulkan: Initial support for context
298abbc156 Roll Chromium from 29bec8631d2f to 9d4a35b46e1e (1220 revisions)
3ca8befb24 Vulkan: Handle multi-context apps in pipeline cache graphs
6607a2b98d Vulkan: Add support for VK_EXT_vertex_input_dynamic_state
bff0b1e43d Change enum value for webgpu to unused value.
6d4706bfb9 WGPU: Add a angle_dawn_dir build override.
aa244358af Reland "Vulkan: Get rid of X11 include in DisplayVkXcb.cpp"
6d589ff6a3 Trace perf: support custom thermal throttling for tests
ec2603d69f Fix build in absence of SSE support
b380ed1f98 Vulkan: Add EGL_ANGLE_global_fence_sync
40dfb3a8bd Fix length() translation for clip/cull distance arrays
d6ceac9159 Metal: Add support for binding slices to images
195c142d7e GLSL test for side effects in prune-able loop
e8a3493f80 Initialize DisplayWgpu
2cae27c296 Vulkan: Enable the doubleDepthBiasConstantFactor feature on NV
197beb4de8 Metal: Crash if for loop body is optimized away
dbc6bd9d4e Reland "Vulkan: Fix alignment issues with SecondaryCommandBuffer"
c673c83758 OpenCL/Vulkan: Initial support for platform/device
b8e56d5d6d Fix an assert when overwriting TexImage binding
243f8ad99f Revert "Vulkan: Fix alignment issues with SecondaryCommandBuffer"
7490ad4d79 Roll vulkan-deps from 2cedf06e4cdf to 063ea20a64fc (7 revisions)
d8340c15c7 Roll Chromium from b650d7fcd665 to 29bec8631d2f (631 revisions)
e53270c9ca Vulkan: Fix alignment issues with SecondaryCommandBuffer
e45b2fd89d Vulkan: Implement ANGLE_translated_shader_source
d9665098a3 Do not use hardcoded ".cr.so" extension for android component builds
9d344b5c82 Uniform block reference in constuctors crash
ebc151d514 Roll vulkan-deps from 4985acbd814d to 2cedf06e4cdf (12 revisions)
2cbf6613c1 Roll Chromium from 3009d13b1e1a to b650d7fcd665 (3044 revisions)
cb7d3cc206 Treat clip/cull distance built-ins as having side effects
e784b1ec82 Manual roll dawn
239e8caa44 Capture/Replay: Disallow concurrent ninja processes.
9fd5167e1f Roll vulkan-deps from 3834da2004ec to 4985acbd814d (77 revisions)
eaddd3baa5 Vulkan: use linear chroma filter for ycbcr by default
1d752a10a7 Roll Chromium from c1ca24b91ed5 to 3009d13b1e1a (567 revisions)
275e6f4fc5 D3D: Add multiplanar support to d3d11 glTexSubImage2D
e489dac03a Allow BGRA -> RGBA for glCopyTex[Sub]Image
5d9abeca4a Revert "Suppress VUID-VkGraphicsPipelineCreateInfo-dynamicRendering-06576"
56e5fa804b Mark bison deps as not shipped
58ccdab977 Roll VK-GL-CTS from b3344240e7fc to b9ec0d4bdf99 (8 revisions)
c3d06480e5 Add dawn to IGNORED_DIRECTORIES.
73fa1b08d0 Use -fno-define-target-os-macros for libpng
3ad163d091 Vulkan: Don't attach format features 2 version of AHB structure
475784f5a6 suppress VUID-VkPipelineVertexInputStateCreateInfo-pNext-pNext
c603a4f199 Don't perf warn about ETC1->ETC2 emulation as it is efficient

ef78e57015 Revert "Vulkan: disable warmUpPipelineCacheAtLink for Venus"
Canonical link: https://commits.webkit.org/276989@main
